math/polynomial/polynomial.go

```diff
@@ -53,6 +53,14 @@ func (p Polynomial) Evaluate(x group.Scalar) group.Scalar {
 	return px
 }
 
```

Comment:
Needs a docucommnet. Plus, I think it's worth documenting the fact that this is a deep-copy.

Reply:
done